.evergreen/create-expansions.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Get the current unique version of this checkout.
```

Comment:
This script was copied from Rust driver (https://github.com/mongodb/mongo-rust-driver/blob/main/.evergreen/create-expansions.sh), not from Java driver, as many other scripts.

I think in this brand-new repo, deviating from Java driver repo is fine and is even desriable. For instance, given the scope of the project, copying Java driverr's `.evg.yml` seems overkill and makes for big maintaining headache.
@jyemin , what is your thoughts?

Reply:
I agree with Nathan on this one, generally.  The Java driver has been around for 16 years and has incurred technical debt that is not necessary to take on in a new project just for consistency.  We've already gone in a different direction on this project on multiple other decisions, including minimum Java version, Gradle version, Gradle language, static analysis, Spotless usage, etc.  Most of these are things we eventually would like to do in the Java driver as well, and there are tickets for most of them.

With regard to scripts in separate files, this is considered a best practice because it makes them easier to edit (because your editor will understand `.sh` files), and to execute independently of Evergreen.  Most of the Java driver scripts are like this.

Reply:
I've manually tested the current evergreen stuff and it works including the testing report uploading. It is also based on best practise like using `subprocess` instead of `shell` command (as per https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Best-Practices#subprocessexec). 

I could remove unnecessary stuff from the `create-expansions.sh` further, but this PR is ready to be used. One of my future PR will depend on this soon (fetching MongoDB's metadata).

Reply:
I am not proposing that we copy the entire `.evg.yml` file, and I am not proposing that we inherit tech debt from the driver. My understanding is:

1. Some differences are necessary, for certain technical reasons (we will not run all the same tests, some configuration will change, we will use different technologies and versions, etc.). This seems the primary concern of @jyemin's comment - I agree with this, and am glad we are upgrading.
2. Other differences are due to some best-practice that we are adopting, not present in the driver, but which can (unlike in 1) be trivially applied to the driver. That is, we have identified non-trivial tech debt in the driver, which is trivially addressable. In this case, we should update the driver or create a ticket to do so.
3. Other differences are arguable or strictly preferential, in which case, we should conform to the Java driver and use it as the baseline, rather than the Rust driver, since this will simplify the review and future debugging. 

It is not always obvious whether any change falls under 1, 2, or 3. I can proceed with the review, and leave comments. I have left a partial review where I started doing this.

If point 3 (trivial consistency) is not actually important, please let me know, and I will ignore certain changes that I can confirm do not change anything (like differences in heading style, usage of $var vs ${var}, etc.).

Reply:
I don't think point 3 matters, for evergreen is not a rocket science and we don't need to refer to Java driver to figure out. It is not complicated to my taste, though sometimes it is time consuming to figure out some detail (like path issue). Copying is less preferable to test and error, which will make for more powerful customization based on real understanding.
It is not that complicated. Sometimes it is complicated due to evergreen's implementation details or even imperfections.